### PR TITLE
GRT 0.0.1 (new formula)

### PR DESCRIPTION
**Pull request by nebgnahz**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

The Gesture Recognition Toolkit (GRT) for Real-time machine learning,
developed by Nick Gillian.

### CODE REVIEW

File: Formula/grt.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  def install
+    cd "build"
+    mkdir "build"
+    cd "build"
+    system "cmake", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", ".."
+    system "make", "-j#{Hardware::CPU.cores}"
+    system "make", "install"
+  end
+  test do
```
Comment (mikemcquaid) [inline Formula/grt.rb]: Newline above this line.

File: Formula/grt.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 "56f90a9ffa8b2bf4e5831d39f9e1912879cf032efa667a5237b57f68800a2dda"
+
+  depends_on "cmake" => :build
+
+  def install
+    cd "build"
+    mkdir "build"
+    cd "build"
+    system "cmake", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", ".."
+    system "make", "-j#{Hardware::CPU.cores}"
```
Comment (mikemcquaid) [inline Formula/grt.rb]: You don't need the `-j`; this is done by default.

File: Formula/grt.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  homepage "http://www.nickgillian.com/wiki/"
+  url "https://github.com/nickgillian/grt/archive/v0.0.1.tar.gz"
+  sha256 "56f90a9ffa8b2bf4e5831d39f9e1912879cf032efa667a5237b57f68800a2dda"
+
+  depends_on "cmake" => :build
+
+  def install
+    cd "build"
+    mkdir "build"
+    cd "build"
```
Comment (mikemcquaid) [inline Formula/grt.rb]: Use `mkdir "build" do` block instead

File: Formula/grt.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    system "cmake", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", ".."
```
Comment (mikemcquaid) [inline Formula/grt.rb]: Use `std_cmake_args` here instead of specifying `prefix`

File: Formula/grt.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    system "make", "-j#{Hardware::CPU.cores}"
+    system "make", "install"
+  end
+  test do
+    # A simple test to see if we have GRT setup properly (only compilation).
```
Comment (mikemcquaid) [inline Formula/grt.rb]: Can remove this comment.

File: Formula/grt.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+class Grt < Formula
+  desc "The Gesture Recognition Toolkit (GRT) for Real-time machine learning."
+  homepage "http://www.nickgillian.com/wiki/"
```
Comment (mikemcquaid) [inline Formula/grt.rb]: http://www.nickgillian.com/wiki/pmwiki.php/GRT/GestureRecognitionToolkit ?

**Comment by nebgnahz** [comment]
Fixed based on the comments (I guess no need to reply each individually). I am reproducing my commit message below:

Fixes based on the review.
1. New line before `test do` block.
2. No `-j` in `make`.
3. Use `mkdir [] do` block.
4. Use `std_cmake_args`.
5. Remove comments.
6. Updated the URL.

File: Formula/grt.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+  test do
+    (testpath/"test.cpp").write <<-EOS.undent
+      #include <GRT/GRT.h>
+      int main() {
+        GRT::GestureRecognitionPipeline pipeline;
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.cpp", "-L#{lib}", "-lgrt", "-o", "test"
```
Comment (UniqMartin) [inline Formula/grt.rb]: `test.cpp` looks like a C++ program (and has a matching extension), but you're using the C compiler (`ENV.cc`) to compile the file. Please use `ENV.cxx` instead (to reduce confusion and avoid problems in the future).

File: Formula/grt.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
+class Grt < Formula
+  desc "The Gesture Recognition Toolkit (GRT) for Real-time machine learning."
```
Comment (UniqMartin) [inline Formula/grt.rb]: These are minor stylistic issues, but please:

- Drop `(GRT)`. The formula name `grt` will be printed in virtually any context where this description will appear.
- Why is “Real-time” capitalized? I don't think it should be.
- Drop the full stop at the end.

File: Formula/grt.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  url "https://github.com/nickgillian/grt/archive/v0.0.1.tar.gz"
+  sha256 "56f90a9ffa8b2bf4e5831d39f9e1912879cf032efa667a5237b57f68800a2dda"
+
+  depends_on "cmake" => :build
+
+  def install
+    cd "build"
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make"
```
Comment (UniqMartin) [inline Formula/grt.rb]: Please remove this line. `make` is implied by `make install` and builds before installing unless there are issues with the build system (unlikely if it is generated by CMake).

**Comment by nebgnahz** [comment]
@UniqMartin Thanks for the feedback. All are fixed.

PTAL.

**Comment by UniqMartin** [comment]
Merged in d43a097c9f503410abb784fcf75bd3d784120977. Thank you for this contribution to Homebrew, @nebgnahz! :tada:

**Pull request closed without merging**